### PR TITLE
FHIR-46657

### PR DESCRIPTION
**Pull request by horsetriboulders**
Add medicationAdherence to QI-Core MedicationRequest and override the ADDITIONAL USCDI text, retaining the US Core Medication Adherence Extension and its URL

QA result:
{32E5BD6A-58FF-45AC-92B2-F3C4D07E1814}

[input\profiles\StructureDefinition-qicore-medicationnotrequested.json](https://github.com/HL7/fhir-qi-core/pull/StructureDefinition-qicore-medicationnotrequested.html) Show Validation Information (1)
\profiles\StructureDefinition-qicore-medicationnotrequested | 0 | 0 | 0
[input\profiles\StructureDefinition-qicore-medicationrequest.json](https://github.com/HL7/fhir-qi-core/pull/StructureDefinition-qicore-medicationrequest.html) Show Validation Information (1)
\profiles\StructureDefinition-qicore-medicationrequest | 0 | 0 | 0

### CODE REVIEW

File: input/profiles/StructureDefinition-qicore-medicationnotrequested.json [hunk 1]
```diff
@@ -71,6 +71,26 @@
...
+			"id" : "MedicationRequest.medicationAdherence",
+			"extension" : [
+			  {
+				"url" : "http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-keyelement",
+				"valueBoolean" : true
+			  }
+			],
+			"path" : "MedicationRequest.extension",
+			"sliceName" : "medicationAdherence",
+			"short" : "URL: http://hl7.org/fhir/us/core/StructureDefinition/us-core-medication-adherence",
```
Comment (yanheras) [inline input/profiles/StructureDefinition-qicore-medicationnotrequested.json:84]: This url for us-core-medication-adherence shouldn't be as short, which rendered as plain text. It should be rendered as a hyperlink that take to the us-core-medication-adherence extension.

File: input/profiles/StructureDefinition-qicore-medicationrequest.json [hunk 2]
```diff
@@ -70,7 +70,27 @@
...
+			"id" : "MedicationRequest.medicationAdherence",
+			"extension" : [
+			  {
+				"url" : "http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-keyelement",
+				"valueBoolean" : true
+			  }
+			],
+			"path" : "MedicationRequest.extension",
+			"sliceName" : "medicationAdherence",
+			"short" : "URL: http://hl7.org/fhir/us/core/StructureDefinition/us-core-medication-adherence",
```
Comment (yanheras) [inline input/profiles/StructureDefinition-qicore-medicationrequest.json:83]: same ... this needs not be a short. It needs to be rendered as a hyperlink to the actual us-core-medication-adherence extension.

File: input/pages/changes.md [hunk 3]
```diff
@@ -47,6 +47,11 @@ This page lists the change history for each version of QI-Core.
 1. Add list of QI-Core-flagged elements to the narrative ([FHIR-46030](https://jira.hl7.org/browse/FHIR-46030)) Applied in each profile ([here](profiles.html))
 1. QICore MedicationRequest prohibits reportedBoolean ([FHIR-46040](https://jira.hl7.org/browse/FHIR-46040)) Applied ([here](StructureDefinition-qicore-medicationrequest.html)) 
 1. Update medication[x] content across not done for reason medication actions to assure consistency ([FHIR-46291](https://jira.hl7.org/browse/FHIR-46291)) Applied ([here](StructureDefinition-qicore-medicationnotrequested.html)) and ([here](StructureDefinition-qicore-medicationadministrationnotdone.html)) and ([here](StructureDefinition-qicore-medicationdispensedeclined.html)) 
+1. QICoreCommunicationRequest description text change ([FHIR-46654](https://jira.hl7.org/browse/FHIR-46654)) Applied ([here](StructureDefinition-qicore-task.html))
```
Comment (yanheras) [inline input/pages/changes.md:50]: These new trackers are part of the ballot reconciliations, they need to be added under the STU7 Publication (7.0.0) section instead.

**Review by yanheras — CHANGES_REQUESTED** [review]

File: input/profiles/StructureDefinition-qicore-medicationrequest.json (hunk 2 above)
Comment (horsetriboulders) [reply input/profiles/StructureDefinition-qicore-medicationrequest.json]: The qi-core medicationRequest and medicationNotRequested are derived from US Core MedicationRequest where the medicationAdherence looks like the following. I've made quite an effort to match the appearance and felt good about it. Obviously they are just snake feet.
{DB94E611-3C67-4C4C-B050-B8FA635DA905}

**Pull request merged**